graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -681,21 +684,27 @@ def _compile_output_step(outputs):
         fold_scope_location = output_context['fold']
 
         expression = None
+        existence_check = None
         if fold_scope_location:
-            if optional:
-                raise AssertionError(u'Unreachable state reached, optional in fold: '
```

Comment:
We're still not allowing `@optional` inside a `@fold`, no? I'm confused about why this is deleted.

Reply:
But we might be allowing `@fold` within `@optional`?
The other case should raise other errors.

Reply:
👍 Let's just make sure to test these situations thoroughly.